streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableRepartitionMap.java

```diff
@@ -101,19 +104,24 @@ public void process(final K key, final Change<V> change) {
     private class KTableMapValueGetter implements KTableValueGetter<K, KeyValue<K1, V1>> {
 
         private final KTableValueGetter<K, V> parentGetter;
+        private ProcessorContext context;
 
         KTableMapValueGetter(final KTableValueGetter<K, V> parentGetter) {
             this.parentGetter = parentGetter;
         }
 
         @Override
         public void init(final ProcessorContext context) {
+            this.context = context;
             parentGetter.init(context);
         }
 
         @Override
-        public KeyValue<K1, V1> get(final K key) {
-            return mapper.apply(key, parentGetter.get(key));
+        public ValueAndTimestamp<KeyValue<K1, V1>> get(final K key) {
+            final ValueAndTimestamp<V> valueAndTimestamp = parentGetter.get(key);
+            return ValueAndTimestamp.make(
+                mapper.apply(key, getValueOrNull(valueAndTimestamp)),
+                valueAndTimestamp == null ? context.timestamp() : valueAndTimestamp.timestamp());
```

Comment:
If there was no mapping for the key upstream, should we just return null?

Reply:
Not sure if I can follow. If `valueAndTimestamp == null`, we effectively call ` mapper.apply(key, null))`. If `mapper` returns `null`, `make()` will return `null` (and it does not matter which timestamp we pass into `make` as it's ignored anyway). If `mapper` does not return `null`, we need to put a timestamp: if `valueAndTimestamp` has a timestamp we use it, otherwise we fall back to `context.timestamp()`.

Does this make sense?

Reply:
Yeah, this was my same mistake about the behavior of `make`. Sorry for the confusion.

Reply:
No need to apologize. Reviewing is asking questions and raising concerns :)